### ISSUE TITLE
unsafe: {String,SliceData} cannot be guarded with build tags

### ISSUE THREAD

**Issue by twmb** [golang/go#58554]
If a go.mod uses anything prior to `go 1.20`, a library cannot use `unsafe.SliceData` even if the usage is guarded in a file with Go version build tags (`//go:build go1.20`).

Using `unsafe.SliceData` or `unsafe.String` will print something along the lines of...
```
./go120.go:7:9: unsafe.SliceData requires go1.20 or later (-lang was set to go1.19; check go.mod)
./go120.go:8:9: unsafe.String requires go1.20 or later (-lang was set to go1.19; check go.mod)
```

This restriction did not exist for `unsafe.Slice` -- `unsafe.Slice` could be guarded with build tags.

### Does this issue reproduce with the latest release?

Yes.

### What did you do?

https://go.dev/play/p/fK1zbeRZrwW

### What did you expect to see?

In the above playground link, no error for both Go 1.19 and Go 1.20.

### What did you see instead?

In the above playground link, no error for Go 1.19 but a compile time error for Go 1.20.

**Comment by cuiweixie** [comment]
if you change go version in go.mod to go 1.16, unsafe.Slice report error too.
check this code, https://go.dev/play/p/QIszeqeVWK3.
currently, the compiler do not check 的 tag, it use go version in go.mod.

**Comment by twmb** [comment]
@cuiweixie thank you, something must've been wrong in my local setup last night -- I confirm this also happens for `unsafe.Slice` in a go1.16 module, even when `unsafe.Slice` is guarded by build tags.

**Comment by bcmills** [comment]
This is true of builtins in general too. It's unfortunate but hard to fix.

Note that setting `go 1.20` in your `go.mod` file won't actually prevent previous Go versions from building your module — the `go` version is currently more “Go version needed for full fidelity” rather than “minimum Go version needed” (compare #57001).

**Comment by go101** [comment]
Looks same as https://github.com/golang/go/issues/58342.

**Comment by go101** [comment]
Can this by solved by supporting `go toolchain.version` in go.mod files?

**Comment by bcmills** [comment]
#59033 is probably relevant.

**Comment by mitar** [comment]
I was hit by this as well, exactly with `unsafe.SliceData`. It works with Go 1.21, but not with prior Go versions.

**Comment by bcmills** [comment]
Duplicate of #59033
